src/android/OpenTokAndroidPlugin.java

```diff
@@ -668,26 +721,76 @@ public void onSignalReceived(Session arg0, String arg1, String arg2, Connection
         }
     }
 
+    // archiveListener
+    public void onArchiveStarted(Session session, String id, String name) {
+        JSONObject data = new JSONObject();
+        try {
+            data.put("id", id);
+            data.put("name", name);
+            triggerJSEvent("sessionEvents", "archiveStarted", data);
+        } catch (JSONException e) {
+        }
+    }
+
+    public void onArchiveStopped(Session session, String id) {
+        JSONObject data = new JSONObject();
+        try {
+            data.put("id", id);
+            triggerJSEvent("sessionEvents", "archiveStopped", data);
+        } catch (JSONException e) {
+        }
+    }
+
     // streamPropertiesListener
+    // Old properties.
+    private boolean hasAudio = false;
+    private boolean hasVideo = false;
+    private JSONObject videoDimensions = new JSONObject();
```

Comment:
Right now we only track the last properties of the **last updated stream**, instead of tracking the last properties of each stream.

We should make these into HashMaps to properly track the last properties of each stream. And remove these properties when the stream drops.

```java
// Old values
private HashMap<String, boolean> streamHasAudio;
private HashMap<String, boolean> streamHasVideo;
private HashMap<String, JSONObject> streamVideoDimensions;

// example
public void onStreamHasAudioChanged(Session session, Stream stream, boolean newValue) {
    boolean oldValue = this.streamHasAudio.get(stream.getStreamId())
    this.streamHasAudio.put(stream.getStreamId(), newValue);

    this.onStreamPropertyChanged("hasAudio", newValue, oldValue, stream);
}
```